orchestrator-salt/src/main/resources/salt/salt/idbroker/init.sls

```diff
@@ -0,0 +1,38 @@
+{%- from 'idbroker/settings.sls' import idbroker with context %}
+{%- from 'gateway/settings.sls' import gateway with context %}
+{{ idbroker.knox_data_root }}/security/keystores/signkey.pem:
+  file.managed:
+    - contents_pillar: gateway:signkey
+    - makedirs: True
+    - mode: 644
+
+{{ idbroker.knox_data_root }}/security/keystores/signcert.pem:
+  file.managed:
+    - contents_pillar: gateway:signcert
+    - makedirs: True
+    - mode: 644
+
+# openssl pkcs12 -export -in cert.pem -inkey key.pem -out signing.p12 -name signing-identity -password pass:admin
+# keytool -importkeystore -deststorepass admin1 -destkeypass admin1 -destkeystore signing.jks -srckeystore signing.p12 -srcstoretype PKCS12 -srcstorepass admin -alias signing-identity
+
+knox-create-sign-pkcs12:
+  cmd.runaws/
+    - name: cd {{ idbroker.knox_data_root }}/security/keystores/ && openssl pkcs12 -export -in signcert.pem -inkey signkey.pem -out signing.p12 -name signing-identity -password pass:{{ salt['pillar.get']('gateway:mastersecret') }}
+    - creates: {{ idbroker.knox_data_root }}/security/keystores/signing.p12
+    - output_loglevel: quiet
+
+{{ idbroker.knox_data_root }}/security/keystores/signing.p12:
+  file.managed:
+    - mode: 644
+    - replace: False
+
+knox-create-sign-jks:
+  cmd.run:
+    - name: cd {{ idbroker.knox_data_root }}/security/keystores/ && keytool -importkeystore -deststorepass {{ salt['pillar.get']('gateway:mastersecret') }} -destkeypass {{ salt['pillar.get']('gateway:mastersecret') }} -destkeystore signing.jks -srckeystore signing.p12 -srcstoretype PKCS12 -srcstorepass {{ salt['pillar.get']('gateway:mastersecret') }} -alias signing-identity
```

Comment:
Not sure if we need the same master secret? If we do we need to extend the pillar/top.sls file otherwise the idbroker instance won't be able to access the gateway pillars. 

pillar/top.sls tells the master which minion can access what pillar data. This is to prevent exposing secrets to random compute nodes.